lib/tasks/migrations.rake

```diff
@@ -17,22 +18,74 @@ namespace :db do
     system(%(pg_dump --exclude-table=ar_internal_metadata --no-owner -h #{db_config['host']} -p #{db_config['port']} -U #{db_config['username']} #{db_config['database']} -f db.sql))
 
     # Move old db.sql on S3 to archive folder
-    system("s3cmd --access_key=#{ENV['AWS_ACCESS_KEY']} --secret_key=#{ENV['AWS_SECRET_KEY']} mv #{BACKUP_S3_PATH}/#{VERSION}-db.sql #{ARCHIVE_S3_PATH}/#{VERSION}/#{Time.now.to_i}-db.sql")
+    system("#{S3CMD} mv #{BACKUP_S3_PATH}/#{VERSION}-db.sql #{ARCHIVE_S3_PATH}/#{VERSION}/#{Time.now.to_i}-db.sql")
 
     # Put new db.sql on S3
-    system("s3cmd --access_key=#{ENV['AWS_ACCESS_KEY']} --secret_key=#{ENV['AWS_SECRET_KEY']} put db.sql #{BACKUP_S3_PATH}/#{VERSION}-db.sql")
+    system("#{S3CMD} put db.sql #{BACKUP_S3_PATH}/#{VERSION}-db.sql")
   end
 
   desc "Restore from backup"
   task restore: :environment do
-    puts "Restoring from S3 backup"
+    puts "Restoring from S3 backup..."
 
-    ENV['PGPASSWORD'] = db_config['password']
+    # Get S3 object key for most recent database backup
+    s3_object_key = `#{S3CMD} ls #{BACKUP_S3_PATH}/ | sort | tail -n 1 | awk '{printf "%s", $4}'`
+
+    puts "Downloading #{s3_object_key}..."
+    system("#{S3CMD} get #{s3_object_key}")
+
+    s3_path, filename_gz = parse_file(s3_object_key)
+    unzip(filename_gz)
+    filename = filename_gz[0, filename_gz.index(".gz")]
+
+    unless File.exist?(filename)
+      puts "#{filename} not found, exit"
+      exit
+    end
+
+    drop_db(db_config['host'], db_config['username'], db_config['database'], db_config['port'])
+
+    # Create database
+    system(%(createdb -e #{db_config['database']} --owner=#{db_config['username']} --host=#{db_config['host']} --port=#{db_config['port']} -U #{db_config['username']} --encoding="UTF8")) or abort('FAILED')
+
+    # Restore database using S3 backup
+    system("psql -h #{db_config['host']} -U #{db_config['username']} #{db_config['database']} < #{filename}")
+
+    # File cleanup
+    system("rm #{filename}")
 
-    # Get latest db.sql from S3
-    system("s3cmd --access_key=#{ENV['AWS_ACCESS_KEY']} --secret_key=#{ENV['AWS_SECRET_KEY']} get #{BACKUP_S3_PATH}/#{VERSION}-db.sql db.sql")
+    if Rails.env == 'development'
```

Comment:
I believe you can test `Rails.env.development?` here.
Just a bit of sugar. :)

Reply:
sweet

Reply:
Yup, this is an [`ActiveSupport::StringInquirer`](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/string_inquirer.rb), which is super cool, partly because the implementation is so simple!